Assets/Code/Scripts/SeaWar.cs

```diff
@@ -1568,7 +1568,17 @@ private void NextPlayer()
         {
             // Obtain the next commander who's still playing
             Commander commander;
-            while (!Commanders[commander = IncrementAttackingCommander()].Playing.Value) {}
+            int attempts = 0; // Add a counter to prevent infinite loop
+            while (!Commanders[commander = IncrementAttackingCommander()].Playing.Value && attempts < Commanders.Count)
+            {
+                attempts++;
+            }
+
+            if (attempts == Commanders.Count)
+            {
+                Debug.LogWarning("No commanders are currently playing. Resetting game state.");
```

Comment:
Does this actually reset the game state?

Reply:
Good catch haha not it doesn't. I meant to handle that case gracefully in some way but forgot to come back to it after I saw that adding the above conditional seemed to solve the issue. Do you think resetting the game state in that case would make sense? Or do you think I should just add a todo saying we don't know what to do with that case?

Reply:
I've not root caused why multiple attempts are required, so I can't really say. It might be worth putting in a todo.

Reply:
Done